content/apps/create-app/activate-nodejs.md

```diff
@@ -1,38 +1,54 @@
 ```js [activate:NodeJS]
 
-params // data from your request handling
-storage // your own memory system
-
-// Retrieve GET query params from your own framework / http handler
-const { pim_url: pimUrl } = params;
-
-// Retrieve your app's Client ID with your own system
-const clientId = storage.get("CLIENT_ID");
-
-// Set the access scopes, take care of the 254 chars max !
-const scopes = 'read_products write_products'; 
-
-// The activate URL should have the pim_url param
-if (!pimUrl) {
-    // Return a Bad request response via your own framework / http server
-    return response(502, { message: "Bad request" });
+import crypto from 'crypto';
+
+// Express.js controller https://expressjs.com
```

Comment:
Don't use the controller concept here: you don't need it.
I suggest to you to define routes directly:

```
const express = require('express')
const app = express()

app.get('/activate', (req, res, next) => {
  //...your code
})
```
Have a look to https://expressjs.com/fr/starter/hello-world.html

Reply:
done